test/suite/test_hs32.py

```diff
@@ -0,0 +1,215 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+# test_hs32.py
+# Ensure that updates without timestamps clear the history store records.
+class test_hs32(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=50MB,statistics=(all)'
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column-fix', dict(key_format='r', value_format='8t')),
+        ('integer-row', dict(key_format='i', value_format='S')),
+        ('string-row', dict(key_format='S', value_format='S')),
+    ]
+    update_type_values = [
+        ('deletion', dict(update_type='deletion')),
+        ('update', dict(update_type='update'))
+    ]
+
+    scenarios = make_scenarios(format_values, update_type_values)
+    nrows = 10000
+
+    def create_key(self, i):
+        if self.key_format == 'S':
+            return str(i)
+        return i
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def evict_cursor(self, uri, nrows):
+        s = self.conn.open_session()
+        s.begin_transaction()
+        # Configure debug behavior on a cursor to evict the page positioned on when the reset API is used.
+        evict_cursor = s.open_cursor(uri, None, "debug=(release_evict)")
+        for i in range(1, nrows + 1):
+            evict_cursor.set_key(self.create_key(i))
+            evict_cursor.search()
+            evict_cursor.reset()
+        s.rollback_transaction()
+        evict_cursor.close()
+
+    def test_non_ts_updates_non_globally_visible_tombstone_clears_hs(self):
+        uri = 'table:test_hs32'
+        create_params = 'key_format={},value_format={}'.format(self.key_format, self.value_format)
+        self.session.create(uri, create_params)
+
+        if self.value_format == '8t':
+            value1 = 97
+            value2 = 98
+        else:
+            value1 = 'a' * 500
+            value2 = 'b' * 500
+
+        # Apply a series of updates from timestamps 1-4.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1))
+        cursor = self.session.open_cursor(uri)
+        for ts in range(1, 5):
+            for i in range(1, self.nrows):
+                with self.transaction(commit_timestamp = ts):
+                    cursor[self.create_key(i)] = value1
+
+        # Reconcile and flush versions 1-3 to the history store.
+        self.session.checkpoint()
+        self.evict_cursor(uri, self.nrows)
+
+        # Apply a another update at timestamp 5.
+        for i in range(1, self.nrows):
+            with self.transaction(commit_timestamp = 5):
+                cursor[self.create_key(i)] = value1
+
+        # Start a long running transaction to make tombstone not globally visible.
```

Comment:
We could add a scenario to cover this long-running transaction since the rest of the test is largely the same as the one below.

Reply:
Thanks. Added a scenario to avoid duplicate test code.